bindings.go

```diff
@@ -0,0 +1,277 @@
+	}
+	return nil
+}
+
+func (ctx evpPkeyCtx) derive(out []byte) ([]byte, error) {
+	outLen := C.size_t(len(out))
+	if out == nil {
+		if C.go_openssl_EVP_PKEY_derive(ctx.ptr, nil, &outLen) != 1 {
+			return nil, newOpenSSLError("EVP_PKEY_derive")
+		}
+		out = make([]byte, outLen)
+	}
+	if C.go_openssl_EVP_PKEY_derive(ctx.ptr, base(out), &outLen) != 1 {
+		return nil, newOpenSSLError("EVP_PKEY_derive")
+	}
+	return out[:outLen], nil
+}
+
+func (ctx evpPkeyCtx) setHKDFProps(mode int, md C.GO_EVP_MD_PTR, key []byte, salt []byte, info []byte) error {
+	switch vMajor {
+	case 3:
+		if mode != 0 {
+			if C.go_openssl_EVP_PKEY_CTX_set_hkdf_mode(ctx.ptr, C.int(mode)) != 1 {
+				return newOpenSSLError("EVP_PKEY_CTX_set_hkdf_mode")
+			}
+		}
+		if md != nil {
+			if C.go_openssl_EVP_PKEY_CTX_set_hkdf_md(ctx.ptr, md) != 1 {
+				return newOpenSSLError("EVP_PKEY_CTX_set_hkdf_md")
+			}
+		}
+		if key != nil {
+			if C.go_openssl_EVP_PKEY_CTX_set1_hkdf_key(ctx.ptr, base(key), C.int(len(key))) != 1 {
+				return newOpenSSLError("gEVP_PKEY_CTX_set1_hkdf_key")
+			}
+		}
+		if salt != nil {
+			if C.go_openssl_EVP_PKEY_CTX_set1_hkdf_salt(ctx.ptr, base(salt), C.int(len(salt))) != 1 {
+				return newOpenSSLError("EVP_PKEY_CTX_set1_hkdf_salt")
+			}
+		}
+		if info != nil {
+			if C.go_openssl_EVP_PKEY_CTX_add1_hkdf_info(ctx.ptr, base(info), C.int(len(info))) != 1 {
+				return newOpenSSLError("EVP_PKEY_CTX_add1_hkdf_info")
+			}
+		}
+		return nil
+	case 1:
+		if mode != 0 {
+			if err := ctx.ctrl(-1, C.GO1_EVP_PKEY_OP_DERIVE, C.GO_EVP_PKEY_CTRL_HKDF_MODE, mode, nil); err != nil {
+				return err
+			}
+		}
+		if md != nil {
+			if err := ctx.ctrl(-1, C.GO1_EVP_PKEY_OP_DERIVE, C.GO_EVP_PKEY_CTRL_HKDF_MD, 0, unsafe.Pointer(md)); err != nil {
+				return err
+			}
+		}
+		if key != nil {
+			if err := ctx.ctrl(-1, C.GO1_EVP_PKEY_OP_DERIVE, C.GO_EVP_PKEY_CTRL_HKDF_KEY, len(key), unsafe.Pointer(base(key))); err != nil {
+				return err
+			}
+		}
+		if salt != nil {
+			if err := ctx.ctrl(-1, C.GO1_EVP_PKEY_OP_DERIVE, C.GO_EVP_PKEY_CTRL_HKDF_SALT, len(salt), unsafe.Pointer(base(salt))); err != nil {
+				return err
+			}
+		}
+		if info != nil {
+			if err := ctx.ctrl(-1, C.GO1_EVP_PKEY_OP_DERIVE, C.GO_EVP_PKEY_CTRL_HKDF_INFO, len(info), unsafe.Pointer(base(info))); err != nil {
+				return err
+			}
+		}
+	default:
+		panic(errUnsupportedVersion())
+	}
+	return nil
+}
+
+func (ctx evpPkeyCtx) setTLS1PRFProps(md C.GO_EVP_MD_PTR, secret []byte, seeds ...[]byte) error {
+
+	switch vMajor {
+	case 3:
+		if C.go_openssl_EVP_PKEY_CTX_set_tls1_prf_md(ctx.ptr, md) != 1 {
+			return newOpenSSLError("EVP_PKEY_CTX_set_tls1_prf_md")
+		}
+		if C.go_openssl_EVP_PKEY_CTX_set1_tls1_prf_secret(ctx.ptr,
+			base(secret), C.int(len(secret))) != 1 {
+			return newOpenSSLError("EVP_PKEY_CTX_set1_tls1_prf_secret")
+		}
+		for _, s := range seeds {
+			if C.go_openssl_EVP_PKEY_CTX_add1_tls1_prf_seed(ctx.ptr,
+				base(s), C.int(len(s))) != 1 {
```

Comment:
Remove wrapping here to match the rest?

```suggestion
		if C.go_openssl_EVP_PKEY_CTX_set1_tls1_prf_secret(ctx.ptr, base(secret), C.int(len(secret))) != 1 {
			return newOpenSSLError("EVP_PKEY_CTX_set1_tls1_prf_secret")
		}
		for _, s := range seeds {
			if C.go_openssl_EVP_PKEY_CTX_add1_tls1_prf_seed(ctx.ptr, base(s), C.int(len(s))) != 1 {
```